src/main/java/org/terasology/flexiblepathfinding/PathfinderSystem.java

```diff
@@ -84,20 +73,30 @@ public int requestPath(EntityRef requester, Vector3i target, List<Vector3i> star
         return requestPath(config, callback);
     }
 
+    private void processPath(ConfigWrapper wrapper) {
+        JPSImpl jps = new JPSImpl(wrapper.config);
+        List<Vector3i> path = Lists.newArrayList();
+        try {
+            if (jps.run()) {
+                path = jps.getPath();
+            }
+        } catch (InterruptedException e) {
+            // do nothing
+        }
+        this.completePathFor(wrapper.config.requester);
+        wrapper.callback.pathReady(path, new Vector3i()); // unusued :??
+    }
+
     public int requestPath(JPSConfig config, PathfinderCallback callback) {
         if (config.requester != null && config.requester.exists()) {
             if (entitiesWithPendingTasks.contains(config.requester)) {
                 return -1;
             }
             entitiesWithPendingTasks.add(config.requester);
         }
-
-        if (config.executor == null) {
-            config.executor = workerTaskMaster.getExecutorService();
-        }
-
-        PathfinderTask task = new PathfinderTask(world, config, callback);
-        taskMaster.offer(task);
+        Mono.just(new ConfigWrapper(callback, config))
+            .subscribeOn(GameScheduler.parallel())
+            .subscribe(this::processPath);
```

Comment:
I think this would be equivalent:
```java
GameScheduler.parallel().schedule(() -> processPath(callback, config))
```
trading the ConfigWrapper tuple class for a lambda.